tests/common/watch_test.go

```diff
@@ -95,3 +97,42 @@ func TestWatch(t *testing.T) {
 		})
 	}
 }
+
+func TestWatchRetryNoLeader(t *testing.T) {
```

Comment:
It might be valuable to have non auth version of this test for making sure that the non auth behavior is preserved. What do you think @lavacat ?